examples/graph/gat_node_classification.py

```diff
@@ -0,0 +1,402 @@
+[Cora](https://linqs.soe.ucsc.edu/data)).
+
+
+### References
+
+For more information on GAT see the original paper
+[Graph Attention Networks](https://arxiv.org/abs/1710.10903) and
+[dgl's Graph Attention Networks](https://docs.dgl.ai/en/0.4.x/tutorials/models/1_gnn/9_gat.html).
+"""
+
+"""
+### Install dependencies
+
+(For easier data preparation)
+```
+pip -q install pandas
+```
+
+(For visualization of models)
+```
+pip -q install pydot
+sudo apt-get -qq install graphviz
+```
+"""
+
+"""
+### Import packages
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+import numpy as np
+import pandas as pd
+import os
+import warnings
+
+warnings.filterwarnings("ignore")
+pd.set_option("display.max_columns", 6)
+pd.set_option("display.max_rows", 6)
+
+"""
+## Dataset
+
+The preparation of the [Cora dataset](https://linqs.soe.ucsc.edu/data) follow that of the
+[Node Classification with Graph Neural Networks](https://keras.io/examples/graph/gnn_citations/)
+tutorial. For more details on the dataset and exploratory data analysis see
+[Node Classification with Graph Neural Networks](https://keras.io/examples/graph/gnn_citations/).
+In brief, the Cora dataset consist of two files: `cora.cites` which contains *directed links* (citations) between
+papers; and `cora.content` which contains *features* of the corresponding papers and one
+of seven labels (the *subject* of the paper).
+"""
+
+zip_file = keras.utils.get_file(
+    fname="cora.tgz",
+    origin="https://linqs-data.soe.ucsc.edu/public/lbc/cora.tgz",
+    extract=True,
+)
+
+data_dir = os.path.join(os.path.dirname(zip_file), "cora")
+
+citations = pd.read_csv(
+    os.path.join(data_dir, "cora.cites"),
+    sep="\t",
+    header=None,
+    names=["target", "source"],
+)
+
+papers = pd.read_csv(
+    os.path.join(data_dir, "cora.content"),
+    sep="\t",
+    header=None,
+    names=["paper_id"] + [f"term_{idx}" for idx in range(1433)] + ["subject"],
+)
+
+class_values = sorted(papers["subject"].unique())
+class_idx = {name: id for id, name in enumerate(class_values)}
+paper_idx = {name: idx for idx, name in enumerate(sorted(papers["paper_id"].unique()))}
+
+papers["paper_id"] = papers["paper_id"].apply(lambda name: paper_idx[name])
+citations["source"] = citations["source"].apply(lambda name: paper_idx[name])
+citations["target"] = citations["target"].apply(lambda name: paper_idx[name])
+papers["subject"] = papers["subject"].apply(lambda value: class_idx[value])
+
+"""
+"""
+
+citations
+
+"""
+"""
+
+papers
+
+"""
+### Split dataset
+
+Notice, splits differ somewhat from that of
+[Node Classification with Graph Neural Networks](https://keras.io/examples/graph/gnn_citations/).
+"""
+
+# Obtain random indices
+random_indices = np.random.permutation(range(papers.shape[0]))
+
+# train data: 50%
+train_data = papers.iloc[random_indices[:1354]]
+# valid data: 10%
+valid_data = papers.iloc[random_indices[1354:1625]]
```

Comment:
Use "validation" instead of "valid" everywhere (since "valid" has a different meaning)